qcodes/instrument/base.py

```diff
@@ -74,6 +76,45 @@ def __init__(self, name: str, metadata: Optional[Mapping[Any, Any]] = None) -> N
 
         self.log = get_instrument_logger(self, __name__)
 
+        if self._call_post_init:
+            self.__post_init__(name, metadata)
+
+    def __init_subclass__(cls, **kwargs: Any) -> None:
```

Comment:
I think a comment / a docstring explaining how the init subclass logic works would be nice

Reply:
I have added docstrings here